### PR TITLE
[330] Gallery challenges

### PR DESCRIPTION
**Pull request by mmiedlarz**
trello card:
https://trello.com/c/ASB8TcV3/330-handle-challenges-endpoint-in-apiclient

Note:
API was just sketched, so it doesn't work yet but I implemented it based on documentation from backend devs.

### CODE REVIEW

File: CreatubblesAPIClient/Sources/DAO/GalleryDAO.swift [hunk 1]
```diff
@@ -105,12 +105,18 @@ class GalleryDAO: NSObject, APIClientDAO {
         return requestSender.send(request, withResponseHandler: handler)
     }
 
+    func getChallengeGalleries(_ pagingData: PagingData?, completion: GalleriesClosure?) -> RequestHandler {
+        let request = ChallengesGalleriesRequest(page: pagingData?.page, perPage: pagingData?.pageSize)
+        let handler = GalleriesResponseHandler(completion: completion)
```
Comment (sebastianewak) [inline CreatubblesAPIClient/Sources/DAO/GalleryDAO.swift:110]: Should we be using the same response handler as for the regular galleries?

**Review by sebastianewak — APPROVED** [review]
Code: +1
Func: n\a

Looks good overall, one comment regarding a reused request handler.

File: CreatubblesAPIClient/Sources/DAO/GalleryDAO.swift (hunk 1 above)
Comment (mmiedlarz) [reply CreatubblesAPIClient/Sources/DAO/GalleryDAO.swift:110]: Yes - challenges are just regular galleries with one extra field and relationships

**Review by DaDer — APPROVED** [review]
👍

**Comment by sebastianewak** [comment]
[Handle "Challenges" endpoint in APIClient](https://trello.com/c/ASB8TcV3/330-handle-challenges-endpoint-in-apiclient)

**Pull request merged**